kernel/src/scan/mod.rs

```diff
@@ -244,21 +265,26 @@ impl Scan {
             };
 
             let read_results =
-                parquet_handler.read_parquet_files(&[meta], read_schema.clone(), None)?;
+                parquet_handler.read_parquet_files(&[meta], self.physical_schema.clone(), None)?;
 
-            let read_expression = if have_partition_cols {
+            let read_expression = if self.have_partition_cols
+                || self.column_mapping_mode != ColumnMappingMode::None
+            {
                 // Loop over all fields and create the correct expressions for them
-                let all_fields: Vec<Expression> = all_fields
+                let all_fields: Vec<Expression> = self
+                    .all_fields
                     .iter()
                     .map(|field| match field {
-                        ColumnType::Partition(field) => {
+                        ColumnType::Partition(field_idx) => {
+                            let field = self.logical_schema.fields.get_index(*field_idx).unwrap().1;
```

Comment:
should we leave a comment here why it is safe to `unwrap`? Or handle the `Result`?.

Reply:
yeah, we should probably. i didn't do too much because i wanted to settle on if we should just clone fields or use indexes first.

Reply:
removed this unwrap anyway